azkaban-common/src/main/java/azkaban/server/AzkabanServer.java

```diff
@@ -48,38 +54,71 @@ public static Props getAzkabanProperties() {
-        .ofType(String.class);
-
-    // Grabbing the azkaban settings from the conf directory.
-    Props azkabanSettings = null;
+    logger.info("loading properties...");
+
+    final OptionSpec<String> configDirectory =
+        parser
+            .acceptsAll(Arrays.asList("c", "conf"), "The conf directory for Azkaban.")
+            .withRequiredArg()
+            .describedAs("conf")
+            .ofType(String.class);
+
+    final OptionSpec<String> emailDirectory =
+        parser
+            .acceptsAll(Arrays.asList("e", "email"),
+                "The path to email templates for " + TemplateBasedMailCreator.class.getSimpleName()
+                    + ".")
+            .withOptionalArg()
+            .describedAs("email")
+            .ofType(String.class);
     final OptionSet options = parser.parse(args);
 
+    final String azkabanHome = System.getenv("AZKABAN_HOME");
+
+    // Grabbing the Azkaban settings from the conf directory.
+    final Optional<Path> configPath;
     if (options.has(configDirectory)) {
-      final String path = options.valueOf(configDirectory);
-      logger.info("Loading azkaban settings file from " + path);
-      final File dir = new File(path);
-      if (!dir.exists()) {
-        logger.error("Conf directory " + path + " doesn't exist.");
-      } else if (!dir.isDirectory()) {
-        logger.error("Conf directory " + path + " isn't a directory.");
-      } else {
-        azkabanSettings = loadAzkabanConfigurationFromDirectory(dir);
-      }
+      final String confHome = options.valueOf(configDirectory);
+      logger.info("Conf parameter set, attempting to get value from " + confHome);
+      configPath = azkabanPath("configuration", confHome);
     } else {
-      logger
-          .info("Conf parameter not set, attempting to get value from AZKABAN_HOME env.");
-      azkabanSettings = loadConfigurationFromAzkabanHome();
+      logger.info("Conf parameter not set, attempting to get value from AZKABAN_HOME=" +
+        azkabanHome +", DEFAULT_CONF_PATH=" + Constants.DEFAULT_CONF_PATH + ".");
+      configPath = azkabanPath("configuration", azkabanHome, Constants.DEFAULT_CONF_PATH);
     }
-
-    if (azkabanSettings != null) {
-      updateDerivedConfigs(azkabanSettings);
+    final Optional<Props> azkabanSettings = configPath.map(path ->
+        loadConfigurationFromPath(path));
+
+    final Optional<Path> emailPath;
+    if (options.has(emailDirectory)) {
+      final String emailHome = options.valueOf(emailDirectory);
+      logger.info("Email parameter set, attempting to get value from " + emailHome);
+      emailPath = azkabanPath("email template", emailHome);
+    } else {
+      logger.info("Email parameter not set, attempting to get value from AZKABAN_HOME=" +
+          azkabanHome +", DEFAULT_EMAIL_TEMPLATE_PATH=" + Constants.DEFAULT_EMAIL_TEMPLATE_PATH + ".");
+      emailPath = azkabanPath("email template", azkabanHome, Constants.DEFAULT_EMAIL_TEMPLATE_PATH);
```

Comment:
Consider simplify some logging messages? Info like Constants.DEFAULT_CONF_PATH and Constants.DEFAULT_EMAIL_TEMPLATE_PATH  might not have to be printed out every time.

Reply:
This message is logged during the loading of properties, which happens during the initialization of the `AzkabanServer`. I am updating the PR.